sdk/core/core-https/src/util/parseCAEChallenges.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * parseCAEChallenges Parses multiple challenges into an array of objects.
+ * Allows users to specify the challenge type through the TChallenge type parameter.
+ */
+export function parseCAEChallenges<TChallenge>(challenges: string): TChallenge[] {
+  if (!challenges) return [{} as TChallenge];
+
+  // Parses a `key="value"` string into an object with { key: "key", value: "value" }
+  const parseKeyValue = (keyValue: string): { key: string; value: string } =>
+    (keyValue.match(/(?<key>\w+(?==))="(?<value>[^"]*)"/) as any)?.groups || {};
```

Comment:
The `as any` is specifically because TypeScript doesn't have RegExp groups in the types. On the other hand, moving away from reg-exes would make this way bigger, the size of it all might make it as equally hard to slip in bugs.

I have provided a detailed test that I wrote to go through all the possible cases.

I also wonder if we could put this parseCAEChallenge outside of core-https. I could keep it in Key Vault until somebody else needed it. However, removing it would make the default parse of the challenge unavailable: implementations would need to provide a challenge parser.